yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -73,7 +73,8 @@ private[spark] class ApplicationMaster(
       } else {
         sparkConf.get(EXECUTOR_INSTANCES).getOrElse(0)
       }
-    val defaultMaxNumExecutorFailures = math.max(3, 2 * effectiveNumExecutors)
+    val defaultMaxNumExecutorFailures = math.max(3,
```

Comment:
Can you add a comment here? Says `effectiveNumExecutors` is `Int.MaxValue` with dynamic allocation enabled  by default?